src/procedures/proc_pagerank.c

```diff
@@ -0,0 +1,149 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "proc_pagerank.h"
+#include "../value.h"
+#include "../util/arr.h"
+#include "../query_ctx.h"
+#include "../util/rmalloc.h"
+#include "../graph/graphcontext.h"
+#include "../algorithms/pagerank.h"
+
+// CALL algo.pageRank('Page', 'LINKS', {iterations: 20, dampingFactor: 0.85}) YIELD node, score
+// CALL algo.pageRank('Page', 'LINKS') YIELD node, score
+
+typedef struct {
+	int n;                          // Number of nodes to rank.
+	int i;                          // Current node to return.
+	Graph *g;                       // Graph.
+	Node node;                      // Node.
+	GrB_Index *mappings;            // Mappings between extracted matrix rows and node ids.
+	LAGraph_PageRank *rankings;     // Nodes rankings.
+	SIValue *output;                // Array with 4 entries ["node", node, "score", score].
+} PagerankContext;
+
+ProcedureResult Proc_PagerankInvoke(ProcedureCtx *ctx, const char **args) {
+	if(array_len(args) != 2) return PROCEDURE_ERR;
+	const char *label = args[0];
+	const char *relation = args[1];
+
+	Schema *s = NULL;
+	GrB_Matrix l = NULL;
+	GrB_Matrix r = NULL;
+	Graph *g = QueryCtx_GetGraph();
```

Comment:
This can just be retrieved from `gc->g`.

Reply:
I prefer not to dig into internals